### ISSUE TITLE
Update to GNAT Community 2020

### ISSUE THREAD

**Issue by jklmnn** [Componolit/ada-runtime#74]
Test and adapt the runtime to GNAT Community 2020

**Comment by jklmnn** [comment]
When compiling the tagged type test with the new compiler the following error comes up:
```
test.adb:19:26: expected type "Object'Class" defined at parent.ads:4
test.adb:19:26: found type "Object" defined at child.ads:6
parent.ads:4:09: run-time library configuration error
```

This test works when compiled with the standard runtime. There don't seem to be any relevant updates in the provided ZFP, though.

**Comment by jklmnn** [comment]
The problem is that `Ada.Tags.Max_Predef_Prims` is initialized to 9 until GCC 8 and to 10 in GCC 9. This value has to be either exactly 9 or 10 depending on the GCC version. Importing this as a symbol from C (where the compiler version could be checked) doesn't work since the error occurs before linking.
A fix would be to change the build process to resolve this symbol already when compiling the runtime. It is currently compiled into a static library that does not resolve symbols between different objects. This could be fixed by merging all object files into a single one with
```
ld -relocatable *.o -o libgnat.o
```
and then creating a static library with a single object file.

**Comment by jklmnn** [comment]
It seems that even if the `libgnat.a` is compiled in a way that the symbol is resolved earlier the compiler doesn't accept it.

**Comment by jklmnn** [comment]
Even withing the constant from another unit doesn't seem to work. The only working variant I can see now would be two different `a-tags.ads` files from different directories. Otherwise we would have to drop GCC 8 but I don't see that as an option.

**Comment by senier** [comment]
Can we at least select the right directory automatically depending on the compiler version?

**Comment by jklmnn** [comment]
> Can we at least select the right directory automatically depending on the compiler version?

This is definitely a goal I'm trying to achieve.

In the mean time I had a better idea. In the standard runtime different versions of `system.ads` are named `system-<platform>.ads`. Instead of directories we could include different versions of `a-tags.ads` named `a-tags-compiler.ads`. Since we copy the files into the include directory anyway before compiling, make could select the correct file depending on the compiler version (and platform) to copy to `adainclude/a-tags.ads`.

**Comment by jklmnn** [comment]
Furthermore with GNAT Community 2020 the STM32F0 GPIO driver doesn't prove anymore with the following message:
```
high: object with constraints on bit representation is unsuitable for aliasing via address clause [possible explanation: type should have an Object_Size representation clause or aspect]
```
I'm not sure if we can fix this without removing the overlaid arrays and using code duplication instead.

**Comment by jklmnn** [comment]
Fixed by #77.
